avocado/plugins/runners/tap.py

```diff
@@ -53,23 +55,24 @@ def _get_tap_result(stdout):
             elif isinstance(event, TapParser.Test):
                 if event.result == TestResult.FAIL:
                     result = "fail"
+                    fail_reason = event.explanation
                     break
                 elif event.result == TestResult.SKIP:
                     result = "skip"
                     break
                 elif event.result == TestResult.XPASS:
                     result = "warn"
+                    fail_reason = f"TODO test {event.name} unexpectedly passed."
```

Comment:
Just one minor thing that I found out: `event.name` is almost always empty, and thus this comes out like:

```
TODO test  unexpectedly passed.
```

Something like:

```diff
diff --git a/avocado/plugins/runners/tap.py b/avocado/plugins/runners/tap.py
index 90193896b..239ab5667 100644
--- a/avocado/plugins/runners/tap.py
+++ b/avocado/plugins/runners/tap.py
@@ -62,7 +62,11 @@ class TAPRunner(ExecTestRunner):
                     break
                 elif event.result == TestResult.XPASS:
                     result = "warn"
-                    fail_reason = f"TODO test {event.name} unexpectedly passed."
+                    if event.name:
+                        tap_test_id = f"{event.number} ({event.name})"
+                    else:
+                        tap_test_id = f"{event.number}"
+                    fail_reason = f"TODO test {tap_test_id} unexpectedly passed."
                     break
                 else:
                     result = "pass"
```

Would do the trick.